dali/test/python/test_operator_one_hot.py

```diff
@@ -66,37 +107,85 @@ def one_hot(input):
+        outputs, reference, batch_size, max_allowed_error=0, expected_layout=expected_layout)
 
 
 def test_one_hot_scalar():
     np.random.seed(42)
     for i in range(10):
-        premade_batch = np.random.randint(0, num_classes, size=batch_size, dtype=np.int32)
-        yield check_one_hot_operator, premade_batch
+        premade_batch = random_scalars_batch()
+        yield partial(check_one_hot_operator, initial_layout=""), premade_batch
 
 
 def test_one_hot_legacy():
     np.random.seed(42)
-    for i in range(10):
-        premade_batch = [np.array([np.random.randint(0, num_classes)], dtype=np.int32)
-                         for x in range(batch_size)]
-        yield check_one_hot_operator, premade_batch, None
+    for j in range(1, 5):  # test 1..4 levels of nested 'multi-dimensional' scalars
+        layout = get_initial_layout(j)
+        for i in range(5):
+            premade_batch = random_scalar_like_tensors_batch(j)
+            yield partial(check_one_hot_operator, axis=None, expected_output_dim=1, initial_layout=layout), premade_batch
 
 
-def test_one_hot ():
+def test_one_hot():
     np.random.seed(42)
+    layout = get_initial_layout(3)
     for i in range(10):
         for axis in [-1, 0, 1, 2, 3]:
-            premade_batch = [
-                np.random.randint(
-                    0, num_classes, size=np.random.randint(2, 8, size=(3,)),
-                    dtype=np.int32) for _ in range(batch_size)]
-            yield check_one_hot_operator, premade_batch, axis
+            expected_layout = modify_layout(layout, axis)
+            premade_batch = random_3d_tensors_batch()
+            yield partial(check_one_hot_operator, initial_layout=layout), premade_batch, axis
+
+
+def test_multi_dim_one_hot_no_initial_layout():
+    np.random.seed(42)
+    for axis in [-1, 0, 1, 2, 3]:
+        premade_batch = random_3d_tensors_batch()
+        yield partial(check_one_hot_operator, axis=axis, initial_layout=""), premade_batch
+
+
+def test_one_hot_reset_layout():
+    np.random.seed(42)
+    layout = get_initial_layout(3)
+    for axis in [-1, 0, 1, 2, 3]:
+        premade_batch = random_3d_tensors_batch()
+        yield partial(check_one_hot_operator, axis=axis, initial_layout=layout, layout_axis_name=""), premade_batch
+    premade_batch = random_scalars_batch()
+    yield partial(check_one_hot_operator, initial_layout="", layout_axis_name=""), premade_batch
+    premade_batch = random_scalar_like_tensors_batch(3)
+    yield partial(check_one_hot_operator, axis=None, expected_output_dim=1, initial_layout=layout, layout_axis_name=""), premade_batch
+
+
+def test_one_hot_custom_layout_axis_name():
+    np.random.seed(42)
+    layout = get_initial_layout(3)
+    premade_batch = random_3d_tensors_batch()
+    yield partial(check_one_hot_operator, axis=-1, initial_layout=layout, layout_axis_name="X"), premade_batch
+    yield partial(check_one_hot_operator, axis=-1, initial_layout=layout, layout_axis_name=0), premade_batch
```

Comment:
I don't like this conversion from int to string here, I would suggest using explicit string.